poc/duplex_sponge.sage

```diff
@@ -64,25 +73,22 @@ class DuplexSponge(DuplexSpongeInterface):
     def __init__(self, iv: bytes):
         assert len(iv) == 32
         self.absorb_index = 0
-        self.squeeze_index = 0
+        self.squeeze_index = self.permutation_state.R
         self.rate = self.permutation_state.R
         self.capacity = self.permutation_state.N - self.permutation_state.R
 
     def absorb(self, input: bytes):
         self.squeeze_index = self.rate
 
         while len(input) != 0:
-            if self.absorb_index == self.rate:
+            if self.absorb_index < self.rate:
+                self.permutation_state[self.absorb_index] = input[0]
+                self.absorb_index += 1
+                input = input[1:]
```

Comment:
can you do this in chunks? I understand that the academic paper does it one by one, but here you're wasting clocks